localized_fields/fields/localized_uniqueslug_field.py

```diff
@@ -49,36 +45,88 @@ def deconstruct(self):
         kwargs['include_time'] = self.include_time
         return name, path, args, kwargs
 
-    def pre_save(self, instance, add: bool):
-        """Ran just before the model is saved, allows us to built
-        the slug.
+    def formfield(self, **kwargs):
+        """Gets the form field associated with this field.
+
+        Because this is a slug field which is automatically
+        populated, it should be hidden from the form.
+        """
+
+        defaults = {
+            'form_class': forms.CharField,
+            'required': False
+        }
+
+        defaults.update(kwargs)
+
+        form_field = super().formfield(**defaults)
+        form_field.widget = forms.HiddenInput()
+
+        return form_field
+
+    def contribute_to_class(self, cls, name, *args, **kwargs):
+        """Hook that allow us to operate with model class. We overwrite save()
+        method to run retry logic.
+
+        Arguments:
+            cls:
+                Model class.
+
+            name:
+                Name of field in model.
+        """
+        # apparently in inheritance cases, contribute_to_class is called more
+        # than once, so we have to be careful not to overwrite the original
+        # save method.
+        if not hasattr(cls, '_orig_save'):
+            cls._orig_save = cls.save
+            max_retries = getattr(
+                settings,
+                'LOCALIZED_FIELDS_MAX_RETRIES',
+                100
+            )
```

Comment:
Hint: We could define the `max_retries` argument into `__init__` (with default value), and pass it (if need) for each field separately. More flexibility and do not uses global setting